tests/unit/test_reservation.py

```diff
@@ -0,0 +1,224 @@
+from typing import List
+from unittest.mock import Mock
+
+import pytest
+
+from ni_measurementlink_service._internal.stubs import session_pb2
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.sessionmanagement.v1 import (
+    session_management_service_pb2,
+)
+from ni_measurementlink_service.session_management import (
+    MultiSessionReservation,
+    SessionInformation,
+)
+from tests.utilities import fake_driver
+
+
+def test___single_session_info___create_session___yields_session_info(
+    session_management_client: Mock,
+) -> None:
+    reservation = MultiSessionReservation(
+        session_management_client, _create_grpc_session_infos(1, "nifake")
+    )
+
+    with reservation.create_session(_construct_session, "nifake") as session_info:
+        assert session_info.session_name == "MySession0"
+        assert session_info.resource_name == "Dev0"
+        assert session_info.instrument_type_id == "nifake"
+
+
+def test___single_session_info___create_session___session_created_and_closed(
+    session_management_client: Mock,
+) -> None:
+    reservation = MultiSessionReservation(
+        session_management_client, _create_grpc_session_infos(1, "nifake")
+    )
+
+    with reservation.create_session(_construct_session, "nifake") as session_info:
+        assert isinstance(session_info.session, fake_driver.Session)
```

Comment:
The setup seems to be identical for the test before but the asserts are slightly different. This one has an assert for 'isinstance'. The previous has an assert on the 'session_name'. Is there a reason for them to be different? Maybe include all the assertions on both tests?

Reply:
"Tests should verify one thing."